src/Native/System.Net.Security.Native/pal_gssapi.h

```diff
@@ -41,14 +41,14 @@ enum PAL_GssFlags : uint32_t
 
 struct PAL_GssBuffer
 {
-    uint64_t length;
+    size_t length;
     uint8_t* data;
 };
 
 /*
 Shims the gss_release_buffer method.
 */
-extern "C" void NetSecurityNative_ReleaseGssBuffer(void* buffer, uint64_t length);
+extern "C" void NetSecurityNative_ReleaseGssBuffer(void* buffer, size_t length);
```

Comment:
If the input to this function is 64bit unconditionally (architecture-independent), in contrary to my first guess in the comments #7090, we need to do "static-cast" there because the native library (krb5-devel, providing gssapi.h) is using size_t, which is 32b in 32b systems.